plugins/modules/zabbix_regexp.py

```diff
@@ -0,0 +1,330 @@
+    state:
+        description:
+            - State of the regular expression.
+        type: str
+        choices: ['present', 'absent']
+        default: 'present'
+
+
+notes:
+    - Only Zabbix >= 6.0 is supported.
+
+extends_documentation_fragment:
+- community.zabbix.zabbix
+
+'''
+
+EXAMPLES = '''
+- name: Update regexp of 'File systems for discovery'
+  community.zabbix.zabbix_regexp:
+    server_url: "{{ zabbix_api_server_url }}"
+    login_user: "{{ zabbix_api_login_user }}"
+    login_password: "{{ zabbix_api_login_pass }}"
+    name: File systems for discovery
+    test_string: ext2
+    expressions:
+      - expression: "^(btrfs|ext2|ext3|ext4|reiser|xfs|ffs|ufs|jfs|jfs2|vxfs|hfs|apfs|refs|ntfs|fat32|zfs)$"
+        expression_type: result_is_true
+'''
+
+RETURN = '''
+msg:
+    description: The result of the operation
+    returned: success
+    type: str
+    sample: 'Successfully updated regular expression setting'
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ansible_collections.community.zabbix.plugins.module_utils.base import ZabbixBase
+from ansible_collections.community.zabbix.plugins.module_utils.version import LooseVersion
+
+import ansible_collections.community.zabbix.plugins.module_utils.helpers as zabbix_utils
+
+
+class RegularExpression(ZabbixBase):
+    def __init__(self, module, zbx=None, zapi_wrapper=None):
+        super(RegularExpression, self).__init__(module, zbx, zapi_wrapper)
+        if LooseVersion(self._zbx_api_version) < LooseVersion('6.0'):
+            module.fail_json(msg="This module doesn't support Zabbix versions lower than 6.0")
+
+    def get_regexps(self, regexp_name):
+        try:
+            regexps = self._zapi.regexp.get(
+                {
+                    "output": "extend",
+                    "selectExpressions": [
+                        "expression",
+                        "expression_type",
+                        "exp_delimiter",
+                        "case_sensitive"
+                    ],
+                    "filter": {
+                        'name': regexp_name
+                    }
+                }
+            )
+            if len(regexps) >= 2:
+                self._module.fail_json("Too many regexps are matched.")
+            return regexps
+        except Exception as e:
+            self._module.fail_json(msg="Failed to get regular expression setting: %s" % e)
+
+    def delete_regexp(self, regexp):
+        try:
+            parameter = [regexp['regexpid']]
+            if self._module.check_mode:
+                self._module.exit_json(changed=True)
+            self._zapi.regexp.delete(parameter)
+            self._module.exit_json(changed=True, msg="Successfully deleted regular expression setting.")
+        except Exception as e:
+            self._module.fail_json(msg="Failed to delete regular expression setting: %s" % e)
+
+    def _compare_expressions(self, current_expressions, future_expressions):
```

Comment:
I wonder if https://github.com/ansible-collections/community.zabbix/blob/a3df0f1a9704085d232980491a29229df2d80e90/plugins/module_utils/helpers.py#L113 or https://github.com/ansible-collections/community.zabbix/blob/a3df0f1a9704085d232980491a29229df2d80e90/plugins/module_utils/helpers.py#L147 wouldn't fit here. What do you think?

Reply:
@D3DeFi 
I apologize for taking much times.
Your suggestion is great to me! I use the function.
In addition, I have modified documents. Will you check commits?
By the way, which version will add this module? I need to change `version_added`.